src/rust/catcollar/futures/accept.rs

```diff
@@ -64,46 +68,63 @@ impl AcceptFuture {
     }
 }
 
-//==============================================================================
+//======================================================================================================================
 // Trait Implementations
-//==============================================================================
+//======================================================================================================================
 
 /// Future Trait Implementation for Accept Operation Descriptors
 impl Future for AcceptFuture {
     type Output = Result<RawFd, Fail>;
 
     /// Polls the underlying accept operation.
     fn poll(self: Pin<&mut Self>, ctx: &mut Context<'_>) -> Poll<Self::Output> {
-        let self_: &AcceptFuture = self.get_mut();
-        match socket::accept(self_.fd as i32) {
+        let self_: &mut AcceptFuture = self.get_mut();
+        match unsafe {
+            let mut address_len: libc::socklen_t = 0;
```

Comment:
Why is this zero?  Shouldn't this be sizeof sockaddr_in?  Assuming we actually want the address filled in, of course.

Reply:
You are correct. I have fixed this. Thanks for checking this.

Reply:
Great, thanks!